db/sqlinterfaces.c

```diff
@@ -1699,9 +1677,6 @@ static void sql_update_usertran_state(struct sqlclntstate *clnt)
     if (!sql)
         return;
 
-    while (isspace(*sql))
-        ++sql;
-
```

Comment:
UNRELATED: I strongly feel, we should offload all the processing of user commands to the parser instead.

Reply:
We make copy of the sql string and use that for hash lookup and so on. Since this only cleans up the leading whitespace this is not too bad, however any difference in spacing inside the query will not match (thus we may need to s/  / /g and that is more of a parser's job).